pkg/apis/sources/v1alpha1/aws_sqs_types.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/knative/pkg/apis/duck"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AwsSqsSource is the Schema for the AWS SQS API
+// +k8s:openapi-gen=true
+// +kubebuilder:categories=all,knative,eventing,sources
+type AwsSqsSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AwsSqsSourceSpec   `json:"spec,omitempty"`
+	Status AwsSqsSourceStatus `json:"status,omitempty"`
+}
+
+// Check that AwsSqsSource can be validated and can be defaulted.
+var _ runtime.Object = (*AwsSqsSource)(nil)
+
+// Check that AwsSqsSource implements the Conditions duck type.
+var _ = duck.VerifyType(&AwsSqsSource{}, &duckv1alpha1.Conditions{})
+
+// AwsSqsSourceSpec defines the desired state of the source.
+type AwsSqsSourceSpec struct {
+	// QueueUrl of the SQS queue that we will poll from.
+	QueueUrl string `json:"queueUrl"`
```

Comment:
1. Better to support both queue URL and queue ARN,
2. Region can be found either URL or ARN, does not need a specific configuration.

Reply:
Agree re. region. I'll fix that.

Re. ARN/URL I agree although for practical purposes the URL seems a better default choice given that the AWS API (https://docs.aws.amazon.com/sdk-for-go/api/service/sqs/#ReceiveMessageInput) takes URLs, so having ARN as an input would require extra transformation.

I'm happy to add it, although in the interest of keeping the PR's size contained I'd do that in a later PR. What do you think?

Reply:
(2) done https://github.com/knative/eventing-sources/pull/124/commits/a69666da0c2073fd2b4a17dafd07a7f23264bc0f

Reply:
I'll resolve this one since there seems to be no objection to adding ARN support later.